pydy/bodies.py

```diff
@@ -0,0 +1,158 @@
+            self._masscenter = Point(self.name + '_masscenter')
+        else:
+            self._masscenter = masscenter
+
+        if mass is None:
+            self._mass = Symbol(self.name + '_mass')
+        else:
+            self._mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(self.name + '_frame')
+        else:
+            self._frame = frame
+
+        if inertia is None and mass is None:
+            inertia = outer(self._frame.x, self._frame.x)  # tensor product
+            self._inertia = (inertia, self._masscenter)
+        else:
+            self._inertia = inertia
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if inertia is None and mass is not None:
+            self.body = Particle(self.name, self._masscenter, self._mass)
+        else:
+            self.body = RigidBody(self.name, self._masscenter, self._frame,
+                                  self._mass, self._inertia)
+
+    def add_coordinate(self, coordinate):
+        self._coordinates.append(coordinate)
+
+    def add_speed(self, speed):
+        self._speeds.append(speed)
+
+    def add_force(self, point_vector, force_vector):
+        """
+        Adds force to the body by adding Force's instance to the force_list.
+        force_list is used by system to get the tuples from the instance and
+        create force_list for Kane's method.
+
+        Parameters
+        ---------
+        point_vector: A 3 Tuple
+            Defines an arbitary point on a body on which the force will be
+            applied. The tuple defines the x, y and z values of the point
+            w.r.t body's masscenter.
+        force_vector: A 3 Tuple
+            Defines the force vector w.r.t frame of the body. The tuple defines
+            the values of vector in x, y and z directions of body's frame.
+
+        Example
+        --------
+        To add a force of magnitude 1 in y direction on a point at distance of
+        1 in x direction. All the directions are w.r.t body's frame.
+
+        >>> body = Body('body')
+        >>> body.add_force((1,0,0), (0,1,0))
+
+        """
+        point_vector = self._convert_tuple_to_vector(point_vector)
+        force_vector = self._convert_tuple_to_vector(force_vector)
+        point = self._masscenter.locatenew(self.name + '_point' + str(self._counter),
+                                           point_vector)
+        self.force_list.append((point, force_vector))
+        self._counter += 1
+
+    def _convert_tuple_to_vector(self, pos_tuple):
+        if len(pos_tuple) == 3:
+            unit_vectors = [self._frame.x, self._frame.y, self._frame.z]
+            vector = Vector(0)
+            for i in range(3):
+                vector += pos_tuple[i] * unit_vectors[i]
+            return vector
+        else:
+            raise TypeError('position tuple must be of length 3')
```

Comment:
Its better to place checks at the starting of the calling method, rather than at the very end, as it reduces overhead.